packages/nfts/contracts/trailblazers-badges/BadgeChampions.sol

```diff
@@ -0,0 +1,251 @@
+        uint256 startTime; // league starts (requires admin action)
+        uint256 seed;
+    }
+
+    mapping(uint256 leagueId => League league) public leagues;
+    uint256 public currentLeagueId = 0;
+
+    TrailblazersBadges public season1Badges;
+    TrailblazersBadgesS2 public season2Badges;
+
+    uint256[8] public powerLevels = [
+        4, // Ravers
+        58, // Robots
+        40, // Bouncers
+        43, // Masters
+        99, // Monks
+        51, // Androids
+        42, // Drummers
+        77 // Shinto
+    ];
+
+    function getChampionId(
+        uint256 _leagueId,
+        address _owner,
+        address _badgeContract,
+        uint256 _tokenId
+    )
+        public
+        pure
+        returns (bytes memory)
+    {
+        return abi.encodePacked(_owner, _badgeContract, _tokenId, _leagueId);
+    }
+
+    event LeagueCreated(
+        uint256 indexed leagueId, uint256 openTime, uint256 startTime, uint256 endTime
+    );
+    event LeagueStarted(uint256 leagueId, uint256 seed);
+
+    event ChampionRegistered(
+        uint256 indexed leagueId,
+        address indexed owner,
+        address badgesContract,
+        uint256 tokenId,
+        uint256 power,
+        uint256 badgeId
+    );
+
+    error ELEMENT_NOT_FOUND();
+    error TOURNAMENT_NOT_STARTED();
+    error TOURNAMENT_NOT_OPEN();
+    error TOURNAMENT_NOT_CLOSED();
+    error TOURNAMENT_NOT_ENDED();
+    error CHAMPION_NOT_OWNED();
+    error INVALID_PARTICIPANT_COUNT();
+    error INVALID_ROUND();
+    error INVALID_CHAMPION_CONTRACT();
+    error INVALID_MATCH();
+
+    function initialize(address _season1Badges, address _season2Badges) external initializer {
+        __Context_init();
+        _grantRole(DEFAULT_ADMIN_ROLE, _msgSender());
+        _transferOwnership(_msgSender());
+        season1Badges = TrailblazersBadges(_season1Badges);
+        season2Badges = TrailblazersBadgesS2(_season2Badges);
+    }
+
+    function getCurrentLeague()
+        public
+        view
+        returns (uint256 openTime, uint256 closeTime, uint256 startTime, uint256 seed)
+    {
+        return getLeague(currentLeagueId);
+    }
+
+    function getLeague(uint256 _leagueId)
+        public
+        view
+        returns (uint256 openTime, uint256 closeTime, uint256 startTime, uint256 seed)
+    {
+        League memory league = leagues[_leagueId];
+        return (league.openTime, league.closeTime, league.startTime, league.seed);
+    }
+
+    function createLeague(
+        uint256 _openTime,
+        uint256 _closeTime,
+        uint256 _startTime
+    )
+        public
+        onlyOwner
+    {
+        League memory league =
+            League({ openTime: _openTime, closeTime: _closeTime, startTime: _startTime, seed: 0 });
+        currentLeagueId += 1;
+
+        leagues[currentLeagueId] = league;
+
+        emit LeagueCreated(currentLeagueId, _openTime, _closeTime, _startTime);
+    }
+
+    modifier leagueOpen(uint256 _leagueId) {
```

Comment:
I think all modifiers shall go before functions, can you lint your code and fix all issues?